docs/kubernetes_setup.md

```diff
@@ -0,0 +1,188 @@
+# Setting up apiserver-network-proxy in Kubernetes
+
+This example will set up apiserver-network-proxy server as a Deployment in the master and apiserver-network-proxy agent as a DaemontSet on the worker nodes.
+
+## Prerequisites
+
+In order to set up the network-proxy, it's needed to generate 2 CA. One for the master communication, `konnectivity-server-ca.crt`
+and one for the cluster communication, `konnectivity-cluster-ca.crt`.
+
+Then, create:
+- A client authentication certificate for the network-proxy agent, `konnectivity-agent-client.crt` and `konnectivity-agent-client.key`.
+- A server authentication certificate for the cluster side of the network-proxy server, `konnectivity-cluster.crt` and `konnectivity-cluster.key` (one for each node or one per node).
+- A client authentication certificate for the Kubernetes API server, `konnectivity-server-client.crt` and `konnectivity-server-client.key`.
+- A server authentication certificate for the server side of the network-proxy server, `konnectivity-server.crt` and `konnectivity-server.key`.
+
+## Master setup
+
+### Kubernetes API server configuration
+
+The Kubernetes API server needs to be started with the following flag: `--egress-selector-config-file=/path/to/your/egress-selector-configuration.yaml`.
+The `egress-selector-configuration.yaml` file looks like:
+
+```yaml
+apiVersion: apiserver.k8s.io/v1alpha1
+kind: EgressSelectorConfiguration
+egressSelections:
+- name: cluster
+  connection:
+    type: http-connect
+    httpConnect:
+      # Here we use 127.0.0.1 since we are binding the apiserver-network-proxy ports on the host
+      # It's possible to use a service instead
+      url: https://127.0.0.1:8131
+      caBundle: /path/to/konnectivity-server-ca.crt
+      clientKey: /path/to/konnectivity-server-client.key
+      clientCert: /path/to/konnectivity-server-client.crt
+- name: master
+  connection:
+    type: direct
+- name: etcd
+  connection:
+    type: direct
+```
+
+### apiserver-network-proxy server
+
+To run the `apiserver-network-proxy` server, use the following Deployment:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
```

Comment:
Is the `konnectivity-server` pod running on the master? It seems like just runs the pod on a random node

Reply:
Yep seems like it 🤔 I'll try to update it this week